.github/workflows/unit_testing.yaml

```
@@ -0,0 +1,31 @@
+name: CMake
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
```

AFAIK this will run the unit tests when the PR is merged into the master branch. Since the PR should have already passed unit tests this is redundant (it's small number of cycles here, but I point it out in case it comes up again).

```suggestion
  pull_request:
```

---

Oh, good call!